server/src/main/java/org/elasticsearch/search/aggregations/support/CoreValuesSourceType.java

```diff
@@ -362,11 +363,44 @@ public void visitLeaf(Query query) {
                             };
                         });
                     }
+                    log.trace("Bounds after query bound date rounding: {}, {}", range[0], range[1]);
 
                     if (range[0] == Long.MIN_VALUE && range[1] == Long.MAX_VALUE) {
                         // Didn't find any bounds
+                        log.trace("Unable to find rounding bounds");
                         return Rounding::prepareForUnknown;
                     }
+
+                    // If we have bounds stepping over each other from query bound checks, return a trivial rounding.
+                    if (range[0] > range[1]) {
+                        return rounding -> new Rounding.Prepared() {
+                            @Override
+                            public long round(long utcMillis) {
+                                return 0;
+                            }
+
+                            @Override
+                            public long nextRoundingValue(long utcMillis) {
+                                return 0;
+                            }
+
+                            @Override
+                            public double roundingSize(long utcMillis, Rounding.DateTimeUnit timeUnit) {
+                                return 0;
+                            }
+
+                            @Override
+                            public double roundingSize(Rounding.DateTimeUnit timeUnit) {
+                                return 0;
+                            }
+
+                            @Override
+                            public long[] fixedRoundingPoints() {
+                                return null;
+                            }
+                        };
```

Comment:
I think maybe we should move this into `Rounding` and give it a couple of unit tests. Maybe just make it a constant?

Reply:
I've convinced myself if we use the unknown rounding, the performance shouldn't matter since it should never get any actual values to use. Then we can avoid having this weird "empty" rounding class that I'm not sure fulfills whatever contracts Rounding is currently guaranteeing.